src/engine/Match_search_mode.ml

```diff
@@ -531,6 +531,8 @@ let children_explanations_of_xpat (env : env) (xpat : Xpattern.t) : ME.t list =
 (* Metavariable condition evaluation *)
 (*****************************************************************************)
 
+let hook_pro_entropy_analysis: (string -> bool) option ref = ref None
```

Comment:
Is it perhaps better to have a `(metavar_analysis_kind -> string -> bool) ref`? Not sure how realistic this is, but if we may reasonably add more pro-specific metavariable analyses it might be nice not to need to proliferate entry points. I suppose this is fine for now in any case, but we should probably consider once n > 1.

Reply:
I fundamentally agree with this - happy to refactor once we have e2e mvp functionality working.